homeassistant/components/amcrest/camera.py

```diff
@@ -536,115 +538,123 @@ def _change_setting(
+            in (
+                await self._api.async_command(
+                    "configManager.cgi?action=getConfig&name=LightGlobal"
+                )
             ).content.decode()
         )
 
-    def _set_indicator_light(self, enable: bool) -> None:
-        self._api.command(
+    async def _set_indicator_light(self, enable: bool) -> None:
+        await self._api.async_command(
             f"configManager.cgi?action=setConfig&LightGlobal[0].Enable={str(enable).lower()}"
         )
 
-    def _change_light(self) -> None:
+    async def _change_light(self) -> None:
         """Enable or disable indicator light."""
-        self._change_setting(
+        await self._change_setting(
             self._audio_enabled or self.is_streaming, "indicator light"
         )
 
-    def _get_motion_recording(self) -> bool:
-        return self._api.is_record_on_motion_detection()
+    async def _get_motion_recording(self) -> bool:
+        return await self._api.async_is_record_on_motion_detection()
 
-    def _set_motion_recording(self, enable: bool) -> None:
-        self._api.motion_recording = enable
+    async def _set_motion_recording(self, enable: bool) -> None:
+        await self._api.async_set_motion_recording(enable)
 
-    def _enable_motion_recording(self, enable: bool) -> None:
+    async def _enable_motion_recording(self, enable: bool) -> None:
         """Enable or disable motion recording."""
-        self._change_setting(enable, "motion recording", "_motion_recording_enabled")
+        await self._change_setting(
+            enable, "motion recording", "_motion_recording_enabled"
+        )
 
-    def _goto_preset(self, preset: int) -> None:
+    async def _goto_preset(self, preset: int) -> None:
         """Move camera position and zoom to preset."""
         try:
-            self._api.go_to_preset(preset_point_number=preset)
+            await self._api.async_go_to_preset(preset_point_number=preset)
         except AmcrestError as error:
             log_update_error(
                 _LOGGER, "move", self.name, f"camera to preset {preset}", error
             )
 
-    def _get_color_mode(self) -> str:
-        return _CBW[self._api.day_night_color]
+    async def _get_color_mode(self) -> str:
+        return _CBW[await self._api.async_day_night_color]
 
-    def _set_color_mode(self, cbw: str) -> None:
-        self._api.day_night_color = _CBW.index(cbw)
+    async def _set_color_mode(self, cbw: str) -> None:
+        await self._api.async_set_day_night_color(_CBW.index(cbw), channel=0)
 
-    def _set_color_bw(self, cbw: str) -> None:
+    async def _set_color_bw(self, cbw: str) -> None:
         """Set camera color mode."""
-        self._change_setting(cbw, "color mode", "_color_bw")
+        await self._change_setting(cbw, "color mode", "_color_bw")
 
-    def _start_tour(self, start: bool) -> None:
+    async def _start_tour(self, start: bool) -> None:
```

Comment:
There is a significant amount of code that is still running in the executor

```

    async def async_enable_recording(self) -> None:
        """Call the job and enable recording."""
        await self.hass.async_add_executor_job(self._enable_recording, True)

    async def async_disable_recording(self) -> None:
        """Call the job and disable recording."""
        await self.hass.async_add_executor_job(self._enable_recording, False)

    async def async_enable_audio(self) -> None:
        """Call the job and enable audio."""
        await self.hass.async_add_executor_job(self._enable_audio, True)

    async def async_disable_audio(self) -> None:
        """Call the job and disable audio."""
        await self.hass.async_add_executor_job(self._enable_audio, False)

    async def async_enable_motion_recording(self) -> None:
        """Call the job and enable motion recording."""
        await self.hass.async_add_executor_job(self._enable_motion_recording, True)

    async def async_disable_motion_recording(self) -> None:
        """Call the job and disable motion recording."""
        await self.hass.async_add_executor_job(self._enable_motion_recording, False)

    async def async_goto_preset(self, preset: int) -> None:
        """Call the job and move camera to preset position."""
        await self.hass.async_add_executor_job(self._goto_preset, preset)

    async def async_set_color_bw(self, color_bw: str) -> None:
        """Call the job and set camera color mode."""
        await self.hass.async_add_executor_job(self._set_color_bw, color_bw)

    async def async_start_tour(self) -> None:
        """Call the job and start camera tour."""
        await self.hass.async_add_executor_job(self._start_tour, True)

    async def async_stop_tour(self) -> None:
        """Call the job and stop camera tour."""
        await self.hass.async_add_executor_job(self._start_tour, False)
```

Reply:
Changed these all over to directly await the underlying async functions.